packages/plot/src/marks/ConnectedMark.js

```diff
@@ -47,13 +40,13 @@ export class ConnectedMark extends Mark {
  * an efficient version with a single scan and the aggregate function
  * argmin and argmax, following https://arxiv.org/pdf/2306.03714.pdf.
  */
-function m4(input, bx, x, y, lo, hi, width, cols = []) {
-  const bins = sql`FLOOR(${width / (hi - lo)}::DOUBLE * (${bx} - ${+lo}::DOUBLE))::INTEGER`;
+function m4(input, bin, x, y, cols = []) {
+  const pixel = sql`FLOOR(${bin})::INTEGER`;
```

Comment:
I know we want to have ints in the end since dealing with decimal is annoying but I think we can omit the integer casting for every sub expression at some point and let the db figure out what the best representation of the types is.

Reply:
In many of these cases the integer casting is helpful. We are binning to a space of pixels, so we have some bounds on the numbers produced. If DuckDB was smarter about generating more compact representations (like small ints) that would be great. But it typically defaults to using long ints which are bigger than we need and also get mapped to BigInts by default by the Arrow lib, which also complicates things. So I suspect we'll want to keep these casts for a while.